File: Week3/94.java
class Solution {
    private List<Integer> ans=new ArrayList<>();
    public List<Integer> inorderTraversal(TreeNode root) {
        inorderTrave(root);
        return ans;
    }
    private void inorderTrave(TreeNode root){
        if (root!=null){
            if (root.left!=null){
                inorderTrave(root.left);
            }
            ans.add(root.val);
            if (root.right!=null){
                inorderTrave(root.right);
            }
        }
    }
}

File: Week3/94.py
'''
https://leetcode-cn.com/problems/binary-tree-inorder-traversal/
非递归解法
'''
class Solution:
    def inorderTraversal(self, root: TreeNode) -> List[int]:
        stack=[]
        res=[]
        cur=root
        while cur or stack:
            while cur:
                stack.append(cur)
                cur=cur.left
            temp=stack.pop()
            res.append(temp.val)
            cur=temp.right
        return res